src/map/status.hpp

```diff
@@ -2693,62 +2693,64 @@ enum e_scs_flag : uint8 {
 };
 
 ///Define flags for the status_calc_bl function. [Skotlex]
-enum e_scb_flag : uint64 {
-	SCB_NONE	= 0x0,
-	SCB_BASE	= 0x00000001,
-	SCB_MAXHP	= 0x00000002,
-	SCB_MAXSP	= 0x00000004,
-	SCB_STR		= 0x00000008,
-	SCB_AGI		= 0x00000010,
-	SCB_VIT		= 0x00000020,
-	SCB_INT		= 0x00000040,
-	SCB_DEX		= 0x00000080,
-	SCB_LUK		= 0x00000100,
-	SCB_BATK	= 0x00000200,
-	SCB_WATK	= 0x00000400,
-	SCB_MATK	= 0x00000800,
-	SCB_HIT		= 0x00001000,
-	SCB_FLEE	= 0x00002000,
-	SCB_DEF		= 0x00004000,
-	SCB_DEF2	= 0x00008000,
-	SCB_MDEF	= 0x00010000,
-	SCB_MDEF2	= 0x00020000,
-	SCB_SPEED	= 0x00040000,
-	SCB_ASPD	= 0x00080000,
-	SCB_DSPD	= 0x00100000,
-	SCB_CRI		= 0x00200000,
-	SCB_FLEE2	= 0x00400000,
-	SCB_ATK_ELE	= 0x00800000,
-	SCB_DEF_ELE	= 0x01000000,
-	SCB_MODE	= 0x02000000,
-	SCB_SIZE	= 0x04000000,
-	SCB_RACE	= 0x08000000,
-	SCB_RANGE	= 0x10000000,
-	SCB_REGEN	= 0x20000000,
+enum e_scb_flag : uint8 {
+	SCB_NONE = 0,
+	SCB_BASE,
+	SCB_MAXHP,
+	SCB_MAXSP,
+	SCB_STR,
+	SCB_AGI,
+	SCB_VIT,
+	SCB_INT,
+	SCB_DEX,
+	SCB_LUK,
+	SCB_BATK,
+	SCB_WATK,
+	SCB_MATK,
+	SCB_HIT,
+	SCB_FLEE,
+	SCB_DEF,
+	SCB_DEF2,
+	SCB_MDEF,
+	SCB_MDEF2,
+	SCB_SPEED,
+	SCB_ASPD,
+	SCB_DSPD,
+	SCB_CRI,
+	SCB_FLEE2,
+	SCB_ATK_ELE,
+	SCB_DEF_ELE,
+	SCB_MODE,
+	SCB_SIZE,
+	SCB_RACE,
+	SCB_RANGE,
+	SCB_REGEN,
 
 	// 4th Job T.Stat/T.Sub-Stat Flags
-	SCB_MAXAP	= 0x40000000,
-	SCB_POW		= 0x80000000,
-	SCB_STA		= 0x000100000000,
-	SCB_WIS		= 0x000200000000,
-	SCB_SPL		= 0x000400000000,
-	SCB_CON		= 0x000800000000,
-	SCB_CRT		= 0x001000000000,
-	SCB_PATK	= 0x002000000000,
-	SCB_SMATK	= 0x004000000000,
-	SCB_RES		= 0x008000000000,
-	SCB_MRES	= 0x010000000000,
-	SCB_HPLUS	= 0x020000000000,
-	SCB_CRATE	= 0x040000000000,
+	SCB_MAXAP,
+	SCB_POW,
+	SCB_STA,
+	SCB_WIS,
+	SCB_SPL,
+	SCB_CON,
+	SCB_CRT,
+	SCB_PATK,
+	SCB_SMATK,
+	SCB_RES,
+	SCB_MRES,
+	SCB_HPLUS,
+	SCB_CRATE,
 
 	// Extra Flags
-	// These are flags not sent through battle/all flags. Always keep these last.
-	SCB_DYE		= 0x080000000000, // force cloth-dye change to 0 to avoid client crashes.
+	SCB_DYE, // force cloth-dye change to 0 to avoid client crashes.
 
-	// Special flags for updating all stat/sub-stat stuff on request.
-	SCB_BATTLE	= 0x07FFFFFFFFFE, // All except BASE and extra flags.
-	SCB_ALL		= 0x07FFFFFFFFFF, // All except extra flags.
-	SCB_MAX		= 0x0FFFFFFFFFFF,
+	SCB_MAX,
+};
+
+// Special flags for updating all stat/sub-stat stuff on request.
+enum e_scb_flag_special : uint64 {
+	SCB_BATTLE = 0x7FFFFFFFFFE, // All except BASE and extra flags.
+	SCB_ALL = 0x7FFFFFFFFFF, // All except extra flags.
```

Comment:
SCB_ALL could be moved directly after SCB_MAX so that it does not affect the bitset size and given proper handling when being parsed during loadtime?